src/Grids/conformal_cubed_sphere_grid.jl

```diff
@@ -0,0 +1,69 @@
+# FIXME? What is the cubed sphere topology?
+# Not sure we can define a {TX, TY, TZ} topology for it?
+
+struct ConformalCubedSphereGrid{FT, F, C}
+                faces :: F
+    face_connectivity :: C
+end
+
+function ConformalCubedSphereGrid(FT=Float64; face_size, z, radius=R_Earth)
+    @warn "ConformalCubedSphereGrid is experimental: use with caution!"
+
+    # +z face (face 1)
+    z⁺_face_grid = ConformalCubedSphereFaceGrid(FT, size=face_size, z=z, radius=radius, rotation=nothing)
+
+    # +x face (face 2)
+    x⁺_face_grid = ConformalCubedSphereFaceGrid(FT, size=face_size, z=z, radius=radius, rotation=RotX(π/2))
+
+    # +y face (face 3)
+    y⁺_face_grid = ConformalCubedSphereFaceGrid(FT, size=face_size, z=z, radius=radius, rotation=RotY(π/2))
+
+    # -x face (face 4)
+    x⁻_face_grid = ConformalCubedSphereFaceGrid(FT, size=face_size, z=z, radius=radius, rotation=RotX(-π/2))
+
+    # -y face (face 5)
+    y⁻_face_grid = ConformalCubedSphereFaceGrid(FT, size=face_size, z=z, radius=radius, rotation=RotY(-π/2))
+
+    # -z face (face 6)
+    z⁻_face_grid = ConformalCubedSphereFaceGrid(FT, size=face_size, z=z, radius=radius, rotation=RotX(π))
+
+    faces = (
+        z⁺_face_grid,
+        x⁺_face_grid,
+        y⁺_face_grid,
+        x⁻_face_grid,
+        y⁻_face_grid,
+        z⁻_face_grid
+    )
+
+    # Construct face connectivity:
+    #
+    # TODO: Use this connectivity as the default
+    # See figure 8.2 of https://mitgcm.readthedocs.io/en/latest/phys_pkgs/exch2.html?highlight=cube%20sphere#generating-topology-files-for-exch2
+    #
+    #         +---+---+
+    #         | 5 | 6 |
+    #     +---+---+---+
+    #     | 3 | 4 |
+    # +---+---+---+
+    # | 1 | 2 |
+    # +---+---+
+    #
+
+    face_connectivity = (
```

Comment:
@ali-ramadhan I don't understand this numbering! Are you sure its correct. I was expecting first row to be

right=2, left=5, above=3, below=6

Reply:
Yes still need to update that. But might move this to a separate PR now!